mautic/plugin-helloworld
# CI init

Docs: https://github.com/mautic/developer-documentation-new/pull/115

This PR is adding a test workflow for GitHub Actions where these checks will run:
- PHPSTAN
- PHPUNIT
- CS Fixer
- Twig Lint
All using config files from Mautic itself to keep the same standard except PHPUNIT.xml where we need different configuration for code coverage.

[The biggest pain](https://community.codecov.com/t/coverage-report-shows-different-files-on-random-changes/4156/4) was to make CodeCov working as the coverage is not for the files in the root dir but in `plugins/HelloWorldBundle`. But now it looks fine:

https://app.codecov.io/github/mautic/plugin-helloworld/tree/ci

I hope that the CodeCov comment bellow will start working once we merge this and it will have a coverage to compare it against.

This may then work as a GitHub Actions template for all the Mautic plugin repositories which should help us with plugin quality.

### Steps to test
1. Check the [Codecov report](https://app.codecov.io/github/mautic/plugin-helloworld/tree/ci)
2. Check the CI builds for this PR
